components/teams-service/storage/postgres/postgres.go

```diff
@@ -390,11 +431,28 @@ func (p *postgres) AddUsers(ctx context.Context,
 // from a different context we should apply project filtering in this function as well.
 // Not doing it to save us a database call since it's not needed currently.
 func (p *postgres) GetUserIDsForTeam(ctx context.Context, id string) ([]string, error) {
+	projectsFilter, err := ProjectsListFromContext(ctx)
+	if err != nil {
+		return nil, p.processError(err)
+	}
+
+	var dbID int
+	var teamProjects []string
+	// ensure the team exists and isn't filtered out by the project filter
+	err = p.db.QueryRowContext(ctx,
+		`SELECT db_id, projects FROM teams
```

Comment:
There must be a way to combine these  two SELECTs 🤔

Reply:
Do we want to use a transaction? We're using the the result from query 1 in query 2, and without a tx the state of the world (or database table) could have changed in between... I think

Reply:
decided to try my hand at an `INNER JOIN` and got it working ✨